pypika/terms.py

```diff
@@ -539,6 +539,8 @@ def __init__(
     ) -> None:
         super().__init__(alias=alias)
         self.name = name
+        if isinstance(table, str):
```

Comment:
Shall we also type-hint `Table` for the `table` variable? I am not sure what is this `Selectable` data type

Reply:
If we do not want `str`, then it is better to also remove it from type-hinting

Reply:
According to /pypika/queries.py#L116, `Table` is a subclass of `Selectable`.
```python
class Table(Selectable):
    ...
```
So I think type-hint `Selectable` is enough.

Reply:
I think `str` is still useful and provides easy access in many cases.
On the other hand this would seem to be more consistent with methods such as `select`, as they also accept `str`.

Reply:
Thank you for elaborating, I agree

Reply:
I would like to ask one more thing. Could you write a test that accepts both data types and works as expected?

Reply:
当然可以，下周我会补充上这个测试。